src/inspect_ai/tool/_tools/_web_search/_web_search.py

```diff
@@ -82,4 +122,122 @@ async def execute(query: str) -> ToolResult:
             else ("I'm sorry, I couldn't find any relevant information on the web.")
         )
 
-    return execute
+    return (
+        ToolDef(
+            execute,
+            name="web_search",
+            options={"use_internal": True, "internal_options": internal_options},
+        ).as_tool()
+        if use_internal
+        else execute
+    )
+
+
+def _google_none_hack() -> Literal["google"]:
+    """If no config nor provider was set, infer 'google' if the API keys are set."""
+    if maybe_get_google_api_keys():
+        deprecation_warning(
+            "The `google` `web_search` provider was inferred based on the presence of environment variables. Please specify the provider explicitly to avoid this warning."
+        )
+        return "google"
+    else:
+        raise ValueError(
+            "Omitting `provider` is no longer supported. Please specify a `web_search` config explicitly to avoid this error."
+        )
+
+
+def _normalize_config(
```

Comment:
This function does the heavy lifting to support the old style function parameters.